api/src/main/java/edu/cornell/mannlib/vitro/webapp/dynapi/DynamicAPIDocumentation.java

```diff
@@ -1,5 +1,770 @@
+        RPC individualGetRPC = resourceAPI.getRpcOnGet();
+        if (individualGetRPC != null) {
+            Action action = actionPool.get(individualGetRPC.getName());
+            pathItem.setGet(individualGetOperation(action, tag));
+        }
+
+        RPC individualPutAction = resourceAPI.getRpcOnPut();
+        if (individualPutAction != null) {
+            Action action = actionPool.get(individualPutAction.getName());
+            pathItem.setPut(individualPutOperation(action, tag));
+        }
+
+        RPC individualPatchAction = resourceAPI.getRpcOnPatch();
+        if (individualPatchAction != null) {
+            Action action = actionPool.get(individualPatchAction.getName());
+            pathItem.setPatch(individualPatchOperation(action, tag));
+        }
+
+        RPC individualDeleteAction = resourceAPI.getRpcOnDelete();
+        if (individualDeleteAction != null) {
+            Action action = actionPool.get(individualDeleteAction.getName());
+            pathItem.setDelete(individualDeleteOperation(action, tag));
+        }
+
+        return pathItem;
+    }
+
+    private PathItem customRESTActionPathItem(CustomRESTAction customRESTAction, Tag tag) {
+        ActionPool actionPool = ActionPool.getInstance();
+
+        PathItem pathItem = new PathItem();
+
+        RPC targetRPC = customRESTAction.getTargetRPC();
+        if (targetRPC != null) {
+            Action action = actionPool.get(targetRPC.getName());
+            HTTPMethod httpMethod = targetRPC.getHttpMethod();
+
+            if (httpMethod != null) {
+                switch (httpMethod.getName().toUpperCase()) {
+                    case "POST":
+                        pathItem.setPost(customRESTActionPostOperation(action, tag));
+                        break;
+                    case "GET":
+                        pathItem.setGet(customRESTActionGetOperation(action, tag));
+                        break;
+                    case "PUT":
+                        pathItem.setPut(customRESTActionPutOperation(action, tag));
+                        break;
+                    case "PATCH":
+                        pathItem.setPatch(customRESTActionPatchOperation(action, tag));
+                        break;
+                    case "DELETE":
+                        pathItem.setDelete(customRESTActionDeleteOperation(action, tag));
+                        break;
+                    default:
+                        break;
+                }
+            }
+        }
+
+        return pathItem;
+    }
+
+    private PathItem actionPathItem(Action action, Tag tag) {
+        PathItem pathItem = new PathItem();
+
+        pathItem.setPost(actionPostOperation(action, tag));
+
+        return pathItem;
+    }
+
+    private Operation collectionGetOperation(Action action, Tag tag) {
+        Operation operation = new Operation();
+        operation.addTagsItem(tag.getName());
+
+        // No description available per resource API rpc
```

Comment:
Maybe it doesn't not exist in examples, but I believe it is allowed to be added in accordance with the ontology: https://github.com/vivo-project/Vitro/blob/sprint-dynapi-2022-feb-staging/home/src/main/resources/rdf/tbox/filegraph/vitro-dynamic-api.owl#L441

Reply:
https://github.com/vivo-project/Vitro/blob/sprint-dynapi-2022-feb-staging/api/src/main/java/edu/cornell/mannlib/vitro/webapp/dynapi/components/ResourceAPI.java

No description referenced in Java class.

Reply:
Actually, it should be added to https://github.com/vivo-project/Vitro/blob/sprint-dynapi-2022-feb-staging/api/src/main/java/edu/cornell/mannlib/vitro/webapp/dynapi/components/RPC.java, right? We need separated description for PUT, POST, GET and DELETE methods? Anyway, it is not there, and I will add it in the separated PR.